src/firewall/core/io/firewalld_conf.py

```diff
@@ -125,6 +136,17 @@ def default(self):
             return str(v)
         return str(self._default)
 
+    def default_as(self, as_type):
+        if (
+            as_type in (bool, int)
+            and self.key_type is as_type
+            and type(self._default) is as_type
+        ):
+            # Optimize common case. We don't need to first convert to string
+            # and parse.
+            return self._default
+        return self.get_as(self.default, as_type)
```

Comment:
I don't think this function is needed. We control the `config. ` values. Instead we should make sure bools are actually `True`/`False`.

e.g.
```diff
diff --git a/src/firewall/config/__init__.py.in b/src/firewall/config/__init__.py.in
index 12b694e8026f..85bd106c2a8a 100644
--- a/src/firewall/config/__init__.py.in
+++ b/src/firewall/config/__init__.py.in
@@ -122,7 +122,7 @@ FALLBACK_LOCKDOWN = False
 FALLBACK_IPV6_RPFILTER = True
 FALLBACK_INDIVIDUAL_CALLS = False
 FALLBACK_LOG_DENIED = "off"
-FALLBACK_AUTOMATIC_HELPERS = "no"
+FALLBACK_AUTOMATIC_HELPERS = False
 FALLBACK_FIREWALL_BACKEND = "nftables"
 FALLBACK_FLUSH_ALL_ON_RELOAD = True
 FALLBACK_RELOAD_POLICY = "INPUT:DROP,FORWARD:DROP,OUTPUT:DROP"
```

Reply:
Ignore this. I see why you made this change now.